### ISSUE TITLE
Logout from Sandbox not working

### ISSUE THREAD

**Issue by auctifera-josed** [mike4aday/SwiftlySalesforce#69]
I'm trying to logout from a sandbox org and is not working, I'm getting the error

`resourceNotFound`

```swift
salesforce.connectedApp.revoke().then { _ -> () in
    SalesforceStore.shared.login()

    }.catch { error in
        print(error)       
}
```

where `login` is

```swift
public func login(){
    salesforce.identity().then { userInfo -> () in
        self.user = userInfo
        }.catch { error in
            print(error)
    }
}
```

This works perfect in developer or production org, I'm stuck here, I have try everything except reseting the entire iPad and I'm still logged in to this sandbox org.

Note: I even removed the call to `login` and is still not working.

**Comment by mike4aday** [comment]
Hi @auctifera-josed - have you set the "login host" to "test.salesforce.com"? 

You can do that via parameter in `configureSalesforce( )` in the app delegate - use "test.salesforce.com" for `loginHost` instead of the default:

From LoginDelegate.swift:
```swift
public func configureSalesforce(consumerKey: String, callbackURL: URL, loginHost: String = ConnectedApp.defaultLoginHost, userID: String = ConnectedApp.defaultUserID, orgID: String = ConnectedApp.defaultOrgID, version: String = Salesforce.defaultVersion) -> Salesforce {
    let connectedApp = ConnectedApp(consumerKey: consumerKey, callbackURL: callbackURL, loginDelegate: self, loginHost: loginHost, userID: userID, orgID: orgID)
    return Salesforce(connectedApp: connectedApp, version: version)
}
```

**Comment by auctifera-josed** [comment]
Hi @mike4aday, no I haven't. But where and how would I set this if the app needs to support any "login host"? Since people may connect to a production org, sandbox org or using a custom domain.

**Comment by hmuronaka** [comment]
Hi, I had the same problem.
I resolved the problem by changing Resource.swift in Swiftlysalesforce to the code below.
What do you think of my solution?
Do you see any potential problem?

// before
 ```swift
extension Resource {
  func asURLRequest() {
    switch self {   
      ....
      case let .revoke(token, host):
        guard let url = URL(string: "https://\(host)/services/oauth2/revoke") else {
        ...
      case let .refresh(refreshToken, consumerKey, host):
        guard let url = URL(string: "https://\(host)/services/oauth2/token") else {
```

// after 
```swift
extension Resource {
  func asURLRequest() {
    switch self {   
      ....
      case let .revoke(token, host):
        // identityURL.host is "login.salesforce.com" or "test.salesforce.com".
        guard let url = URL(string: "https://\(authData.identityURL.host ?? host)/services/oauth2/revoke") else {
        ...
      case let .refresh(refreshToken, consumerKey, host):
        guard let url = URL(string: "https://\(authData.identityURL.host ?? host)/services/oauth2/token") else {
```

**Comment by mike4aday** [comment]
@hmuronaka the only issue I foresee is that your fix would get overwritten if you update the library. Your fix is great, and I'll incorporate it. There's no need for a `host` associated value for the `revoke` enum. Thanks.

**Comment by hmuronaka** [comment]
Hi, @mike4aday 
Thank you for your reply and your review.
I sent fixed code. https://github.com/mike4aday/SwiftlySalesforce/pull/72

**Comment by mike4aday** [comment]
Fixed in version 6.0.7. Thanks @hmuronaka
